modules/waybar/hm.nix

```
@@ -1,76 +1,62 @@
+      @define-color base04 ${base04}; @define-color base05 ${base05}; @define-color base06 ${base06}; @define-color base07 ${base07};
 
-    @define-color base08 ${base08}; @define-color base09 ${base09}; @define-color base0A ${base0A}; @define-color base0B ${base0B};
-    @define-color base0C ${base0C}; @define-color base0D ${base0D}; @define-color base0E ${base0E}; @define-color base0F ${base0F};
+      @define-color base08 ${base08}; @define-color base09 ${base09}; @define-color base0A ${base0A}; @define-color base0B ${base0B};
+      @define-color base0C ${base0C}; @define-color base0D ${base0D}; @define-color base0E ${base0E}; @define-color base0F ${base0F};
 
-    * {
-        font-family: ${sansSerif.name};
-        font-size: ${builtins.toString sizes.desktop}pt;
-    }
+      * {
+          font-family: ${sansSerif.name};
+          font-size: ${builtins.toString sizes.desktop}pt;
+      }
 
-    window#waybar, tooltip {
-        background: alpha(@base00, ${with config.stylix.opacity; builtins.toString desktop});
-        color: @base05;
-    }
+      window#waybar, tooltip {
+          background: alpha(@base00, ${with config.stylix.opacity; builtins.toString desktop});
+          color: @base05;
+      }
 
-    tooltip {
-        border-color: @base0D;
-    }
+      tooltip {
+          border-color: @base0D;
+      }
     ''
     + (builtins.readFile ./base.css)
-    + (if config.stylix.targets.waybar.enableLeftBackColors then builtins.readFile ./left.css else ''
-    .modules-left #workspaces button {
-        border-bottom: 3px solid transparent;
-    }
-    .modules-left #workspaces button.focused,
-    .modules-left #workspaces button.active {
-        border-bottom: 3px solid @base05;
-    }
-    '')
-    + (if config.stylix.targets.waybar.enableCenterBackColors then builtins.readFile ./center.css else ''
-    .modules-center #workspaces button {
-        border-bottom: 3px solid transparent;
-    }
-    .modules-center #workspaces button.focused,
-    .modules-center #workspaces button.active {
-        border-bottom: 3px solid @base05;
-    }
-    '')
-    + (if config.stylix.targets.waybar.enableRightBackColors then builtins.readFile ./right.css else ''
-    .modules-right #workspaces button {
-        border-bottom: 3px solid transparent;
-    }
-    .modules-right #workspaces button.focused,
-    .modules-right #workspaces button.active {
-        border-bottom: 3px solid @base05;
-    }
-    '');
+    + (if config.stylix.targets.waybar.enableLeftBackColors then (import ./colors.nix { palce = "left"; }) else colorlessModules "left")
+    + (if config.stylix.targets.waybar.enableCenterBackColors then (import ./colors.nix { palce = "center"; }) else colorlessModules "center")
+    + (if config.stylix.targets.waybar.enableRightBackColors then (import ./colors.nix { palce = "right"; }) else colorlessModules "right");
```

Small typo here:

```suggestion
    + (if config.stylix.targets.waybar.enableLeftBackColors then (import ./colors.nix { place = "left"; }) else colorlessModules "left")
    + (if config.stylix.targets.waybar.enableCenterBackColors then (import ./colors.nix { place = "center"; }) else colorlessModules "center")
    + (if config.stylix.targets.waybar.enableRightBackColors then (import ./colors.nix { place = "right"; }) else colorlessModules "right");
```

This could also be simplified if we replace `{place}:` with `place:` in `colors.nix`, then we can write:

```suggestion
    + (if config.stylix.targets.waybar.enableLeftBackColors then import ./colors.nix "left" else colorlessModules "left")
    + (if config.stylix.targets.waybar.enableCenterBackColors then import ./colors.nix "center" else colorlessModules "center")
    + (if config.stylix.targets.waybar.enableRightBackColors then import ./colors.nix "right" else colorlessModules "right");
```

---

Will fix tomorrow!

---

Fixed!